src/tests/test_get_dataset_openml.py

```diff
@@ -0,0 +1,89 @@
+import json
+
+import responses
+from sqlalchemy import Engine
+from sqlalchemy.orm import Session
+from starlette.testclient import TestClient
+
+from database.models import Dataset
+from tests.testutils.paths import path_test_resources
+
+OPENML_URL = "https://www.openml.org/api/v1/json"
+
+
+def test_happy_path(client: TestClient, engine: Engine):
+    dataset_description = Dataset(name="anneal", platform="openml", platform_specific_identifier="1")
+
+    with responses.RequestsMock() as mocked_requests:
+        _mock_normal_responses(mocked_requests, dataset_description)
+        with Session(engine) as session:
+            # Populate database
+            session.add(dataset_description)
+            session.commit()
+        response = client.get("/dataset/1")
```

Comment:
`session` shouldn't interact with the `mocked_requests` as far as I understand, so:

```suggestion
    with Session(engine) as session:
        # Populate database
        session.add(dataset_description)
        session.commit()
    with responses.RequestsMock() as mocked_requests:
        _mock_normal_responses(mocked_requests, dataset_description)
        response = client.get("/dataset/1")
```